frontend/src/components/public_group/DonationDistributor.js

```diff
@@ -0,0 +1,447 @@
+  }
+
+  constructor(props) {
+    super(props);
+    const {group, method, amount} = this.props;
+    this.stripeKey = group.stripeAccount && group.stripeAccount.stripePublishableKey ? group.stripeAccount.stripePublishableKey : '';
+    const hasPaypal = group.hasPaypal;
+    const hasStripe = this.stripeKey && amount;
+    this.state = {
+      disabled: !hasPaypal && !hasStripe,
+      paymentMethod: method,
+      optionalComission: true,
+      includeComission: false,
+      optionalPaymentMethod: hasStripe && hasPaypal
+    };
+    this.resetOptions()
+  }
+
+  renderDonateButton()
+  {
+    return (
+      <div className='DonationDistributor-donate-button max-width-1 mx-auto'>
+        <div 
+          className={`Button ${this.state.disabled ? 'Button--disabled': 'Button--green'}`} 
+          onClick={this.state.disabled ? Function.prototype : this.open.bind(this)}
+        >
+          Donate
+        </div>
+      </div>
+    )
+  }
+
+  renderCollectives()
+  {
+    const options = this.options;
+    return options.map(
+      (option, index) => {
+        return (
+          <DonationDistributorItem 
+            key={index} 
+            label={option.label} 
+            icon={option.icon} 
+            editable={options.length > 1 && option.editable} 
+            value={option.value * 100} 
+            amount={this.getDistributableAmount()}
+            onChange={(percent, prev) => {
+              const delta = Math.abs(prev - percent) / 100;
+              if (prev - percent < 0)
+              {
+                option.value += delta;
+                this.redistribute('remove', option, delta);
+              }
+              else
+              {
+                option.value -= delta;
+                this.redistribute('add', option, delta);
+              }
+            }}
+          />
+        )
+      }
+    )
+  }
+
+  renderSubtotal()
+  {
+    const {currency} = this.props;
+    const subtotal = this.getSubTotal();
+    const formattedSubtotal = formatCurrency(subtotal, currency, {compact: true});
+    return (
+      <div className='flex DonationDistributor-subtotal'>
+        <div className='DonationDistributor-subtotal-label flex-auto left-align'>Sub-total</div>
+        <div className='DonationDistributor-subtotal-amount'>{formattedSubtotal}</div>
+      </div>
+    )
+  }
+
+  renderCommision()
+  {
+    const {currency} = this.props;
+    const {includeComission, optionalComission} = this.state;
+    const comissionAmount = this.getCommissionAmount();
+    const formattedComissionAmount = formatCurrency(comissionAmount, currency, {compact: true});
+    return (
+      <div>
+        <div className='DonationDistributor-method-label'>Commission</div>
+        <div className='DonationDistributorItem-container' style={{marginTop: 0}}>
+          <div className='flex'>
+            <div className='flex-auto'>
+              <div className='flex flex-column'>
+                <div className='flex'>
```

Comment:
Yes, to get the right layout, since it is using flex boxes.
See http://www.basscss.com/#basscss-flexbox for examples